nemo/collections/asr/models/classification_models.py

```diff
@@ -174,7 +174,7 @@ def setup_training_data(self, train_data_config: Optional[Union[DictConfig, Dict
         # Need to set this because if using an IterableDataset, the length of the dataloader is the total number
         # of samples rather than the number of batches, and this messes up the tqdm progress bar.
         # So we set the number of steps manually (to the correct number) to fix this.
-        if 'is_tarred' in train_data_config and train_data_config['is_tarred']:
+        if isinstance(self._train_dl.dataset, torch.utils.data.IterableDataset):
```

Comment:
Train dl can be None. The earlier config check is safer. Or add more checks to your version.

Reply:
Fixed